src/shell/Program.cs

```diff
@@ -1,92 +1,154 @@
+            app.Option<bool>("--stack-trace", "", CommandOptionType.NoValue, o => o.ShowInHelpText = false, inherited: true);
+            app.Conventions
+                .UseDefaultConventions()
+                .UseConstructorInjection(services);
+            ShellExtensions extensions = LoadShellExtensions(services.GetService<IFileSystem>());
+            try
+            {
+                ShellExtension? extension;
+                if (extensions.TryFindCommand(args, out extension) && extension != null)
+                {
+                    var chainFactory = services.GetRequiredService<ExpressChainManagerFactory>();
+                    var txExecutorFactory = services.GetRequiredService<TransactionExecutorFactory>();
+                    var input = chainFactory != null ? chainFactory.GetConnectionFilePath(string.Empty) : string.Empty;
+                    return await extension.ExecuteAsync(args, input, Console.Out, Console.Error, chainFactory, txExecutorFactory);
+                }
+                else
+                {
+                    return await app.ExecuteAsync(args);
+                }
+            }
+            catch (CommandParsingException ex)
+            {
+                await Console.Error.WriteLineAsync($"\x1b[1m\x1b[31m\x1b[40m{ex.Message}");
+
+                if (ex is UnrecognizedCommandParsingException uex && uex.NearestMatches.Any())
+                {
+                    await Console.Error.WriteLineAsync();
+                    await Console.Error.WriteLineAsync("Did you mean this?");
+                    await Console.Error.WriteLineAsync("    " + uex.NearestMatches.First());
+                }
+
+                return 1;
+            }
+            catch (Exception ex)
+            {
+                app.WriteException(ex);
+                return -1;
+            }
         }
 
-        return 1;
-      }
-      catch (Exception ex)
-      {
-        app.WriteException(ex);
-        return -1;
-      }
-    }
+        private static ShellExtensions LoadShellExtensions(IFileSystem? fileSystem)
+        {
+            if (fileSystem == null)
+            {
+                return new ShellExtensions();
+            }
+            string rootPath = fileSystem.Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".neo");
+            string filePath = fileSystem.Path.Combine(rootPath, "extensions.json");
+            if (!fileSystem.File.Exists(filePath))
+            {
+                return new ShellExtensions();
+            }
+            string json = File.ReadAllText(filePath);
+            var extensions = ShellExtensions.FromJson(json);
+            return extensions;
+        }
 
-    private int OnExecute(CommandLineApplication app, IConsole console)
-    {
-      console.WriteLine("You must specify a subcommand.");
-      app.ShowHelp(false);
-      return 1;
-    }
+        private static int ExecuteProcess(string[] args, string cmd, string input)
```

Comment:
is this dead code?

Reply:
yes, thank you! It was from the initial poc